vnext/Microsoft.ReactNative/Views/ReactRootControl.cpp

```diff
@@ -612,16 +627,19 @@ void ReactRootControl::ReloadViewHost() noexcept {
 }
 
 void ReactRootControl::AttachBackHandlers(XamlView const &rootView) noexcept {
+
   /*
-  // If we are running in a Xaml Island or some other environment where the SystemNavigationManager is unavailable,
-  // we should just skip hooking up the BackButton handler.
-  // Unfortunately SystemNavigationManager->GetForCurrentView seems to crash with XamlIslands, not sure how to detect
-  // this right now auto fact = winrt::get_activation_factory< winrt::Windows::UI::Core::SystemNavigationManager,
-      winrt::Windows::UI::Core::ISystemNavigationManagerStatics>();
-  auto abi = fact.as<ABI::Windows::UI::Core::ISystemNavigationManagerStatics>();
-  ABI::Windows::UI::Core::ISystemNavigationManager* sysNavMgr;
-  auto hr = abi->GetForCurrentView(&sysNavMgr);
-  */
+   * If we are running in a Xaml Island or some other environment where the SystemNavigationManager is unavailable,
+   * we should just skip hooking up the BackButton handler.
+   * SystemNavigationManager->GetForCurrentView seems to crash with XamlIslands, so we detect if we are
+   * running in the desktop windowing model before calling it.
+   */
+  AppPolicyWindowingModel e;
+  if (FAILED(AppPolicyGetWindowingModel((HANDLE)-6 /*GetCurrentThreadEffectiveToken()*/, &e)) ||
```

Comment:
does -6 have any meaning? 
should this be using the commented out `GetCurrentThreadEffectiveToken` ?

Reply:
Similar to the other one, this is only exposed in the header for WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM).